Gems/Atom/Tools/AtomToolsFramework/Code/Source/Util/Util.cpp

```diff
@@ -661,6 +662,197 @@ namespace AtomToolsFramework
         return paths;
     }
 
+    AZStd::string GetAbsolutePathForSourceAsset(const AZStd::string& path)
+    {
+        bool found = false;
+        AZ::Data::AssetInfo sourceInfo;
+        AZStd::string watchFolder;
+        AzToolsFramework::AssetSystemRequestBus::BroadcastResult(
+            found, &AzToolsFramework::AssetSystemRequestBus::Events::GetSourceInfoBySourcePath, path.c_str(), sourceInfo, watchFolder);
+
+        if (found)
+        {
+            AZStd::string result;
+            if (AzFramework::StringFunc::Path::ConstructFull(watchFolder.c_str(), sourceInfo.m_relativePath.c_str(), result, true))
+            {
+                return result;
+            }
+        }
+
+        return path;
+    }
+
+    AZStd::vector<AZStd::string> GetPathsForAssetSourceDependencies(const AZ::Data::AssetInfo& sourceInfo)
```

Comment:
An `AZStd::set<AZ::IO::Path>` could be returned here, however an AZStd::set is less cache line friendly.

The benefit is the function doesn't have to sort at the end.
So it is tradeoff to whether you think it would be slower for the user to iterate over a red black tree of Asset Source Paths or the time it takes to sort a vector of Asset Source Paths.

Reply:
Instead of using a set, I sorted the container and removed nonunique values